### ISSUE TITLE
$elemMatch not working properly?

### ISSUE THREAD

**Issue by doender** [crcn/sift.js#203]
In the code below, shouldn't case 2 result in `true`?

```javascript
const match = sift({
    areas: {
        $elemMatch: {
            type: { $eq: 'driveway' },
            elements: {
                $nin: ['grass', 'green']
            }
        }
    },
});

// Case 1
match({
    areas: [
        { type: 'roof' },
        { type: 'driveway', propertyLines: ['green'], elements: ['tiles'] }
    ]
});
// true, as expected

// Case 2
match({
    areas: [
        { type: 'driveway', propertyLines: ['green'], elements: ['tiles'] },
        { type: 'roof' }
    ]
});
// false, but should be true?
```

**Comment by doender** [comment]
I think this is related: 
```
expect(sift({
   moves: {
      $elemMatch: {
        player: {
           $ne: 5
        }
     }
  }
})({ 
  moves: [
    { player: 3 }, { player: 5 }
  ]})
).toBeTrue();
```

This test fails, whereas it works on https://mongoplayground.net/

**Comment by crcn** [comment]
Sorry for not getting around to this sooner, problem should be fixed in `13.2.0`!
